packages/next/build/webpack/loaders/next-swc-loader.js

```diff
@@ -101,6 +101,8 @@ export function pitch() {
   ;(async () => {
     let loaderOptions = this.getOptions() || {}
     if (
+      // TODO: investigate swc file reading in PnP mode?
+      !process.versions.pnp &&
       loaderOptions.fileReading &&
       !EXCLUDED_PATHS.test(this.resourcePath) &&
```

Comment:
It's weird - why isn't `EXCLUDED_PATHS` enough? 🤔 Is `this.resourcePath` the right parameter? Looking at the report in https://github.com/vercel/next.js/issues/31812#issuecomment-1011059472 it shows that the reported path should match:

```
C:\test-nextjs\.yarn\__virtual__\next-virtual-c649e2e81d\0\cache\next-npm-12.0.8-aa55acca00-947f0295aa.zip\node_modules\next\dist\client\dev\amp-dev.js
```

Reply:
Note thart you need to escape the backslashes in the literal for the regex to match:

```
console.log(EXCLUDED_PATHS.test('C:\\test-nextjs\\.yarn\\__virtual__\\next-virtual-c649e2e81d\\0\\cache\\next-npm-12.0.8-aa55acca00-947f0295aa.zip\\node_modules\\next\\dist\\client\\dev\\amp-dev.js'))
```

Reply:
Ah right, hmm yeah I'm not able to find any mis-matches shared in the errors from the threads 🤔